src/components/DataKeyPair.tsx

```diff
@@ -82,6 +84,43 @@ export const DataKeyPair: FC<DataKeyPairProps> = (props) => {
   const enableClipboard = useJsonViewerStore(store => store.enableClipboard)
   const { copy, copied } = useClipboard()
 
+  const enableHighlight = useJsonViewerStore(store => store.enableHighlight)
+  const isHighlight = useMemo(() => {
+    if (!enableHighlight || prevValue === undefined) return false
+    if (typeof value !== typeof prevValue) {
+      return true
+    }
+    if (!Array.isArray(value)) {
+      if (Array.isArray(prevValue)) {
+        return true
+      }
+
+      if (typeof value === 'object' || typeof value === 'function') {
+        return false
+      }
+
+      if (value !== prevValue) {
+        return true
+      }
+    }
+    return false
```

Comment:
This part can be extracted to `utils` with some test

Reply:
wait, this is not a normal compare function... nah ignore what I said

Reply:
ok, i added some comments here